### ISSUE TITLE
Exposing API docs

### ISSUE THREAD

**Issue by ghost** [typings/discussions#16]
Hi there,

I was just thinking ... wouldn't it be an idea to [typedoc](http://typedoc.io/) each registered definition, and add that information to the (yet to be completed) registry? That way, developers can quickly look up the type signature of a function, and see which (sometimes hidden) type names are used.

**Comment by demurgos** [comment]
Most of the definitions I've seen already add comments with the related doc. I think formatting it a bit (adding `@param <nam>` and `@returns`) could be encouraged. Even if there are already lots of definitions, the project is still young so it could be possible to promote it.

I don't really know if it's should be mandatory, but it's definitely looks like a good thing (in the end, both `typedoc` and `typings` want to provide some meta-data about the code).

**Comment by blakeembrey** [comment]
I love the idea. I need to get around to getting `typedoc` updated though. Aside from that, it would be nice to have. Individual definition repos could definitely generate docs and I think registry items would be an awesome 👍 if we build a website 😄

**Comment by ghost** [comment]
Haha indeed if there is a website. Glad you like the idea 😄  I'd love to help - if you need any help - but I probably won't be available until the upcoming summer.

**Comment by blakeembrey** [comment]
@samvv Of course, that would be amazing 😄 I'd love to put together a beautiful site that helps people use TypeScript definitions.

**Comment by SomaticIT** [comment]
👍 
1. I think it could be really helpful.
2. It could create an awesome visibility for typings.

**Comment by SomaticIT** [comment]
Just to let you know :

We built a module which generates `jsdoc` comments based on TypeScript definitions files (`.d.ts`).
It automatically adds types to `jsdoc` and provide a simple template to display correctly special tags.

**ts2jsdoc** | [Website](http://ts2jsdoc.js.org/) | [Source](https://github.com/spatools/ts2jsdoc)
Samples : 
- [node.js typings](http://ts2jsdoc.js.org/node)
- [typescript 1.4 typings](http://ts2jsdoc.js.org/typescript)

The project is outdated (works with TypeScript 1.4) and we looks for volunteer to upgrade the project to TypeScript 1.8+.

Is anyone interested ?

Thanks.

**Comment by ghost** [comment]
@SomaticIT Nope, sorry. I'm sticking with TypeDoc. Wish you good luck with the project, though.
